tests/steamship_tests/utils/test_key_value_store.py

```diff
@@ -0,0 +1,237 @@
+
+
+class KeyValueStore:
+    """A simple key value store implemented in Steamship.
+
+    Each entry is stored as an empty file whose "Key" has:
+      * Kind = KeyValueStore
+      * Name = the key of the (kv) pair
+      * Value = a dict set to the value
+
+    This class is stored inside the `tests` folder because:
+      1) It has proven useful enough in package development that we wish to make sure it's tested, and
+      2) We wish to add KV-functionality to Spaces in general, but
+      3) We're not eager to directly add this as a utility class in the codebase.
+
+    For the time being; consider this a copy-paste reference implementation along with tests.
+    """
+
+    client: Steamship
+    namespace: str
+
+    def __init__(self, client: Steamship, namespace: str = "KeyValueStore"):
+        self.client = client
+        self.namespace = namespace
+
+    def _get_file(self, or_create: bool = False) -> Optional[File]:
+        status_files = File.query(self.client, f'filetag and kind "{self.namespace}"').data.files
+        if len(status_files) == 0:
+            if not or_create:
+                return None
+            return File.create(
+                self.client,
+                blocks=[Block.CreateRequest(text="")],
+                tags=[Tag.CreateRequest(kind=self.namespace, name="__init__")],
+            ).data
+        else:
+            return status_files[0]
+
+    def get(self, key: str) -> Optional[Dict]:
+        """Gets the value represented by `key`."""
+        file = self._get_file()
+
+        if file is None:
+            return None
+
+        for tag in file.tags:
+            if tag.kind == self.namespace and tag.name == key:
+                return tag.value
+
+    def delete(self, key: str) -> bool:
+        """Deletes the entry represented by `key`"""
+        file = self._get_file()
+
+        if file is None:
+            return False
+
+        deleted = False
+        for tag in file.tags:
+            if tag.kind == self.namespace and tag.name == key:
+                tag.delete()
+
+        return deleted
+
+    def set(self, key: str, value: Dict[str, Any]):
+        # Sets the entry (key, value)
+
+        # First delete it if it exists to avoid duplicate tags.
+        self.delete(key)
+
+        # Now get/create the file
+        file = self._get_file(or_create=True)
+
+        req = Tag.CreateRequest(file_id=file.id, kind=self.namespace, name=key, value=value)
+        return self.client.post("tag/create", req, expect=Tag)
+
+    def reset(self):
+        """Deletes all keys"""
+        file = self._get_file()
+        if file is not None:
+            file.delete()
+
+
+class SpecializationStatus(str, Enum):
+    """These are for use with the"""
+
+    UNSPECIALIZED = "unspecialized"
+    SPECIALIZED = "specialized"
+    SPECIALIZATION_IN_PROGRESS = "specialization_in_progress"
+
+    @staticmethod
+    def from_str(string: str) -> "SpecializationStatus":
```

Comment:
If I'm not mistaken, we don't need this method. Instead, you can use the default init method of Enum: `SpecializationStatus("unspecialized")`.